### PR TITLE
Maximum Sum of subarray

### PR DESCRIPTION
**Pull request by MRK04**
## Related Issue
-  Find the sum subarray within 1D array that has maximum sum. #323

## Proposed Changes
- Added Code in LearnCPP/A-Array/MaximumSumSubarray.cpp
- Added readme file describing all about problem statement, different approaches and how to implement it.

## Additional Info
- I have added all three possible solution to solve the problem. Started solving from brute force approach to going to its optimized solution. 

Test Details: \
I have run the test cases on GCC compiler and I run the following cases 

Test Case 1 \
 Input: \
 a[] = {-2, -3, 4, -1, -2, 1, 5, -3} \
 Output: \
 7  

Test Case 2 \
Input: \
 Input: \
 a[] = {-2, 3, -1, 2} \
 Output: \
 4

## Checklist

- [x] ✅ My code follows the code style of this project.
- [ ] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots

| code goes here1  | 
image

| code goes here2 |
image

| output  |
image
image
image
image

### CODE REVIEW

**Comment by MRK04** [comment]
@Lakhankumawat Please review this and let me know if any changes are required.

**Comment by MRK04** [comment]
@miraj0507 or @n4i9kita pls review my code.

**Comment by Lakhankumawat** [comment]
Change your code and README Directory from `A-Array/subArrays/Maximum_sum_of_subArray.cpp` to `A-Array/MaximumSumSubarray.cpp` & README.md

**Comment by Lakhankumawat** [comment]
Uncheck all the checks which do not apply to you

**Comment by Lakhankumawat** [comment]
You need to follow particular guidelines for your README , please follow
these guidelines [click here](https://github.com/Lakhankumawat/LearnCPP#to-add-programs-using-stl-follow-same-approach-as--algorithms) 
also you can see how your README should look like [see here](https://github.com/Lakhankumawat/LearnCPP/blob/main/S-SortingAlgorithms/README.md) 
for more please go through these [here](https://youtu.be/7UgC3HDETRA)

**Comment by MRK04** [comment]
> You need to follow particular guidelines for your README , please follow these guidelines [click here](https://github.com/Lakhankumawat/LearnCPP#to-add-programs-using-stl-follow-same-approach-as--algorithms) also you can see how your README should look like [see here](https://github.com/Lakhankumawat/LearnCPP/blob/main/S-SortingAlgorithms/README.md) for more please go through these [here](https://youtu.be/7UgC3HDETRA)

I have done all the changes as suggested.

**Comment by siddhi-244** [comment]
Hey , you have committed 3 files!

**Comment by MRK04** [comment]
> Hey , you have committed 3 files!

I have corrected all the things. You can review it.

**Comment by MRK04** [comment]
@siddhi-244 pls review my pr.

File: A-Array/MaximumSumSubarray.cpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            currsum=0;
+        }
+        maxSum=max(maxSum,currsum);
+    }
+    return maxSum;
+}
+
+int main() {
+    int a[] = {-2, -3, 4, -1, -2, 1, 5, -3};
+    int n = sizeof(a)/sizeof(a[0]);
```
Comment (siddhi-244) [inline A-Array/MaximumSumSubarray.cpp:72]: Could you take array as input ?

Comment (MRK04) [reply A-Array/MaximumSumSubarray.cpp]: @siddhi-244 I have added the code for taking array as input.

**Review by siddhi-244 — APPROVED** [review]
LGTM

**Review by Lakhankumawat — APPROVED** [review]
LGTM 💯🌟

**Comment by Lakhankumawat** [comment]
@all-contributors please add @MRK04 code ,

**Pull request merged**